lib/matplotlib/dviread.py

```diff
@@ -861,39 +901,48 @@ def _register(self, words):
         # http://tex.stackexchange.com/questions/10826/
         # http://article.gmane.org/gmane.comp.tex.pdftex/4914
 
-        texname, psname = words[:2]
-        effects, encoding, filename = '', None, None
-        for word in words[2:]:
-            if not word.startswith('<'):
-                effects = word
-            else:
-                word = word.lstrip('<')
-                if word.startswith('[') or word.endswith('.enc'):
-                    if encoding is not None:
-                        matplotlib.verbose.report(
-                            'Multiple encodings for %s = %s'
-                            % (texname, psname), 'debug')
-                    if word.startswith('['):
-                        encoding = word[1:]
-                    else:
-                        encoding = word
-                else:
-                    assert filename is None
-                    filename = word
+        # input must be bytestrings (the file format is ASCII)
+        for word in words:
+            assert isinstance(word, bytes)
```

Comment:
This is a private function and should only be called from the same class, so if this triggers it's an internal error in the code. While writing this code I felt this was more obvious than the later errors you get from mixing bytestrings and strings, but perhaps this is not really needed.